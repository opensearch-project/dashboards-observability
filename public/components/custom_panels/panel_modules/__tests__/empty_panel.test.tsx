/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { waitFor } from '@testing-library/react';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import React from 'react';
import { EmptyPanelView } from '../empty_panel';

describe('Empty panel view component', () => {
  configure({ adapter: new Adapter() });

  it('renders empty panel view with disabled popover', async () => {
    const addVizDisabled = true;
    const showFlyout = jest.fn();
    const wrapper = mount(
      <EmptyPanelView addVizDisabled={addVizDisabled} showFlyout={showFlyout} />
    );

    wrapper.update();
    await waitFor(() => {
      expect(
        toJson(wrapper, {
          mode: 'deep',
        })
      ).toMatchSnapshot();
    });
    expect(wrapper.find('EuiButton').prop('disabled')).toBe(true);
  });

  it('renders empty panel view with enabled popover', async () => {
    const addVizDisabled = false;
    const showFlyout = jest.fn();
    const wrapper = mount(
      <EmptyPanelView addVizDisabled={addVizDisabled} showFlyout={showFlyout} />
    );

    wrapper.update();
    await waitFor(() => {
      expect(
        toJson(wrapper, {
          mode: 'deep',
        })
      ).toMatchSnapshot();
    });
    expect(wrapper.find('EuiButton').prop('disabled')).toBe(false);
  });
});
