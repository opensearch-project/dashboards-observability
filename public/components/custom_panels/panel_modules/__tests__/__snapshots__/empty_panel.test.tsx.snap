// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Empty panel view component renders empty panel view with disabled popover 1`] = `
<div>
  <div
    className="euiSpacer euiSpacer--xxl"
  />
  <div
    className="euiText euiText--medium"
    data-test-subj="addFirstVisualizationText"
  >
    <div
      className="euiTextAlign euiTextAlign--center"
    >
      <h2>
        Start by adding your first visualization
      </h2>
      <div
        className="euiSpacer euiSpacer--m"
      />
      <div
        className="euiText euiText--medium"
      >
        <div
          className="euiTextColor euiTextColor--subdued"
        >
          Use PPL Queries to fetch & filter observability data and create visualizations
        </div>
      </div>
    </div>
  </div>
  <div
    className="euiSpacer euiSpacer--m"
  />
  <div
    className="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--justifyContentCenter euiFlexGroup--directionRow euiFlexGroup--responsive"
  >
    <div
      className="euiFlexItem euiFlexItem--flexGrowZero"
    >
      <div
        className="euiPopover euiPopover--anchorDownLeft"
        id="addVisualizationContextMenu"
      >
        <div
          className="euiPopover__anchor"
        >
          <button
            className="euiButton euiButton--primary euiButton-isDisabled"
            data-test-subj="addVisualizationButton"
            disabled={true}
            onClick={[Function]}
            style={
              Object {
                "minWidth": undefined,
              }
            }
            type="button"
          >
            <span
              className="euiButtonContent euiButtonContent--iconRight euiButton__content"
            >
              <svg
                aria-hidden={true}
                className="euiIcon euiIcon--medium euiIcon--inherit euiIcon-isLoading euiButtonContent__icon"
                focusable="false"
                height={16}
                role="img"
                style={null}
                viewBox="0 0 16 16"
                width={16}
                xmlns="http://www.w3.org/2000/svg"
              />
              <span
                className="euiButton__text"
              >
                Add visualization
              </span>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="euiSpacer euiSpacer--xxl"
  />
</div>
`;

exports[`Empty panel view component renders empty panel view with enabled popover 1`] = `
<div>
  <div
    className="euiSpacer euiSpacer--xxl"
  />
  <div
    className="euiText euiText--medium"
    data-test-subj="addFirstVisualizationText"
  >
    <div
      className="euiTextAlign euiTextAlign--center"
    >
      <h2>
        Start by adding your first visualization
      </h2>
      <div
        className="euiSpacer euiSpacer--m"
      />
      <div
        className="euiText euiText--medium"
      >
        <div
          className="euiTextColor euiTextColor--subdued"
        >
          Use PPL Queries to fetch & filter observability data and create visualizations
        </div>
      </div>
    </div>
  </div>
  <div
    className="euiSpacer euiSpacer--m"
  />
  <div
    className="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--justifyContentCenter euiFlexGroup--directionRow euiFlexGroup--responsive"
  >
    <div
      className="euiFlexItem euiFlexItem--flexGrowZero"
    >
      <div
        className="euiPopover euiPopover--anchorDownLeft"
        id="addVisualizationContextMenu"
      >
        <div
          className="euiPopover__anchor"
        >
          <button
            className="euiButton euiButton--primary"
            data-test-subj="addVisualizationButton"
            disabled={false}
            onClick={[Function]}
            style={
              Object {
                "minWidth": undefined,
              }
            }
            type="button"
          >
            <span
              className="euiButtonContent euiButtonContent--iconRight euiButton__content"
            >
              <svg
                aria-hidden={true}
                className="euiIcon euiIcon--medium euiIcon--inherit euiButtonContent__icon"
                focusable="false"
                height={16}
                role="img"
                style={null}
                viewBox="0 0 16 16"
                width={16}
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M13.069 5.157 8.384 9.768a.546.546 0 0 1-.768 0L2.93 5.158a.552.552 0 0 0-.771 0 .53.53 0 0 0 0 .759l4.684 4.61c.641.631 1.672.63 2.312 0l4.684-4.61a.53.53 0 0 0 0-.76.552.552 0 0 0-.771 0Z"
                  fillRule="non-zero"
                />
              </svg>
              <span
                className="euiButton__text"
              >
                Add visualization
              </span>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="euiSpacer euiSpacer--xxl"
  />
</div>
`;
